components/script/dom/document.rs

```diff
@@ -221,6 +221,9 @@ pub struct Document {
     /// For each element that has had a state or attribute change since the last restyle,
     /// track the original condition of the element.
     modified_elements: DOMRefCell<HashMap<JS<Element>, ElementSnapshot>>,
+    /// If paint was suppressed by layout, this tells us whether we need to
+    /// reflow or not.
+    needs_paint: Cell<bool>,
```

Comment:
This comment is a little confusing, because it uses the term "reflow" while the name is talking about "paint". How about something like "This flag will be true if layout suppressed a reflow attempt that was needed in order for the page to be painted."